lightningd/channel.c

```diff
@@ -756,6 +756,14 @@ void channel_fail_permanent(struct channel *channel,
 			    const char *fmt,
 			    ...)
 {
+	/* FIXME: should I compare the whole scid for stub channel?
+		I doubt there are actually any channels in the Genesis block(1x1x1) */
+	if(channel->scid->u64 >> 40 == 1){
```

Comment:
Is `1x1x1` a special value in this case? We're not actually guaranteed to have an `scid` at the time we create the recovery data, since the channel may not have yet confirmed.

Reply:
I see it below, seems a bit strange to use this, and not a way more unreasonable `UINT64_MAX`, but it'd ultimately be the same anyway. Maybe it'd be better to just keep it `NULL`?

Reply:
I am setting it to 1x1x1 to indicate (while loading the in-memory channel from DB) that it's a stub channel and we need to send the error to the peer, nothing else. It was on my to-do list to set it to 0x0x0 (as Rusty suggested), but it won't make any difference. 

I think setting it to NULL will create confusion with the unconfirmed channels which also don't have any scid? If that's not the case I am happy to set it to NULL.

Reply:
I like the idea to make the code unambiguous for the new readers, maybe we can make a `#DEFINE GENESIS_SHORT_ID "1x1x1"`

P.S: I'm assuming that it is easy to convert into a string the `channel->scid`

Reply:
I need to remember to read the PR from bottom to top, the Rusty suggestion is better I think https://github.com/ElementsProject/lightning/pull/5288#discussion_r881234472